.github/workflows/tests.yml

```diff
@@ -0,0 +1,201 @@
+# Unless required by applicable law or agreed to in writing, software is distributed under the License is       
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or              
+# implied.  See the License for the specific language governing permissions and limitations under the License.  
+
+---
+name: Unit Tests
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  build-lib-cpp-tests:
+    strategy:
+      matrix:
+        os: [ windows-latest, macos-latest, ubuntu-latest ]
+    name: Build ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Enable Developer Command Prompt 💻
+        if: runner.os == 'Windows'
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+
+      - name: Setup Valgrind 🔬
+        if: runner.os == 'Linux'
+        run: sudo apt-get -y install valgrind
+
+      - name: Setup Java ☕
+        uses: actions/setup-java@v3.4.1
+        with:
+          distribution: temurin
+          java-version: 8
+          cache: sbt
+
+      - name: Setup Python 🐍
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Conan 💪
+        id: conan
+        uses: turtlebrowser/get-conan@main
+
+      - name: Conan version 💪
+        run: echo "${{ steps.conan.outputs.version }}"
+
+      - name: Setup cmake 🔧
+        uses: lukka/get-cmake@latest
+
+      - name: Prepare, Build, and Install Ωedit 🔧
+        run: |
+          cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS=YES --install-prefix "${PWD}/_install"
+          cmake --build build
+          cmake --install build/packaging --prefix "${PWD}/_install" --config Debug
+
+      - name: Copy library file mac/linux 🚚
+        if: runner.os != 'Windows'
+        run: cp -r _install/lib/* ./lib
+        shell: bash
+
+      - name: Copy library file windows 🚚
+        if: runner.os == 'Windows'
+        run: cp -r _install/bin/omega_edit.dll lib/
+        shell: bash
+
+      - name: Test Ωedit RPC on macOS 📋
+        if: runner.os == 'macOS'
+        run: |
+          build/bin/server_test
+
+      - name: Test Ωedit RPC on Linux 📋
+        if: runner.os == 'Linux'
+        run: |
+          valgrind build/bin/server_test
+
+      - name: Prepare, and Build Ωedit Integration Tests 🔧
+        run: |
+          cmake -S src/tests/integration -B build-tests-integration -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="${PWD}/_install"
+          cmake --build build-tests-integration
+
+      - name: Run Ωedit Integration Tests 📋
+        working-directory: build-tests-integration
+        run: |
+          ctest -C Debug --output-on-failure
+
+      - name: Build Tests 🔧
+        run: |
+          cmake -S src/tests -B build-tests -DCMAKE_BUILD_TYPE=Debug
+
+      - name: Run Tests 📋
+        if: runner.os != 'Windows'
```

Comment:
With the recent refactoring of the build, this test _might_ work on Windows now.

Reply:
Will test this in another different PR. Possibly after getting the TS client tests running